docs/source/development/low-level-api.rst

```diff
@@ -0,0 +1,307 @@
+model's XML tree, while also taking into account fragment boundaries and the
+aforementioned placeholder elements.
+
+.. class:: MelodyLoader
+   :noindex:
+
+   .. automethod:: iterancestors
+      :noindex:
+   .. automethod:: iterchildren_xt
+      :noindex:
+   .. automethod:: iterdescendants
+      :noindex:
+   .. automethod:: iterdescendants_xt
+      :noindex:
+
+Resolving references
+====================
+
+You will often encounter attributes that contain references to other elements.
+
+The ``MelodyLoader`` provides the following methods to work with references:
+
+.. class:: MelodyLoader
+   :noindex:
+
+   .. automethod:: follow_link
+      :noindex:
+   .. automethod:: follow_links
+      :noindex:
+   .. automethod:: create_link
+      :noindex:
+
+Finding elements elsewhere
+==========================
+
+The low-level API implements the fundamentals for looking up model objects or
+finding them by their type. The following methods are involved in these
+operations:
+
+.. class:: MelodyLoader
+   :noindex:
+
+   .. automethod:: iterall
+      :noindex:
+   .. automethod:: iterall_xt
+      :noindex:
+   .. automethod:: xpath
+      :noindex:
+   .. automethod:: xpath2
+      :noindex:
+
+Manipulating objects
+====================
+
+.. warning::
+
+   The low-level API by itself does not do any consistency or validity checks
+   when modifying a model. Therefore it is very easy to break a model using it,
+   which can be very hard to recover from. Proceed with caution.
+
+As ``GenericElement`` instances are simply wrappers around the raw XML
+elements, any changes to their attributes are directly reflected by changes to
+the attributes or children of the underlying XML element and vice versa. This
+means that no special care needs to be taken to keep the high-level and
+low-level parts of the API synchronized.
+
+In many cases, the attribute names of the high-level API match those in the
+XML, with the difference that the former uses ``snake_case`` naming (as is
+conventional in the Python world), while the latter uses ``camelCase`` naming.
+This example shows how the name of a function is accessed and modified using
+the low-level API:
+
+.. code-block:: python
+
+   >>> myfunc = model.search("LogicalFunction")[3]
+   >>> myfunc.name
+   'defend the surrounding area against Intruders'
+   >>> myfunc._element.attrib["name"]
+   'defend the surrounding area against Intruders'
+   >>> myfunc._element.attrib["name"] = "My Function"
+   >>> myfunc.name
+   'My Function'
+
+Be aware that the XML usually does not explicitly store attributes that are set
+to their default value (as defined by the meta model). In addition to that, the
+high-level API often offers convenience shortcuts and reverse lookups that are
+not directly reflected by XML attributes. This means it can be difficult to
+infer the correct attributes for an object when only looking at the high-level
```

Comment:
I think here is a semantical typo. Didn't you mean the low-level API? You praised the convenience shortcuts and reverse lookups for easier access of attributes and their interesting relationships. It doesn't make sense that inference of correct attributes can be difficult.

```suggestion
infer the correct attributes for an object when only looking at the low-level
```

Reply:
I mean that, precisely because the high-level API has these additional shortcuts, lookups, etc., it can be difficult to know what to input at the lower level. Could this be phrased a bit better maybe?

Reply:
Do you mean something like:

> [...] often offers convenience shortcuts, filters and reverse lookups that are
> not directly reflected by XML attributes and may be designed for special use cases.
> Without looking into the rules or declaration in the high-level API it can be difficult to
> infer the correct attributes for an object at the low-level API.

Reply:
Yes, that sounds better indeed. I've pushed a new version, can you take a look at it again?